src/components/ui/drawers/NameEditionDrawer.tsx

```diff
@@ -31,6 +34,8 @@ interface INameEditionDrawer {
   open: boolean;
   setOpen: (value: boolean) => void;
   type: 'sharehouse' | 'category';
+  shareHouseId?: string;
+  categoryId?: string;
```

Comment:
Could you modify this to accept only one ID, which will change dynamically based on the given `type`? For example, if `type` is `sharehouse`, the ID will be interpreted as `shareHouseId`.